accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -545,6 +549,38 @@ per
 
 ## Runtime semantics
 
+### Weak and strong semantics
+
+To allow the null safety feature to be rolled out incrementally, we define two
+modes of execution.
+
+**Weak checking** mode largely ignores the nullability of types at runtime, as
```

Comment:
What checks do you actually get with weak checking? If there are no observable checks from a developer's viewpoint, then calling this 'weak checking' seems a bit confusing.

Reply:
Do you prefer just using "weak mode" and "strong mode"?  I can make a pass to use that uniformly.  The "checking" here is because it is primarily instance checks and casts that are affected by the semantics: in weak mode, instance checks and casts are "weak" in the sense that they ignore nullability, whereas in strong mode they are "strong" in the sense that they properly account for nullability.

Reply:
I'm OK with 'weak checking' then.

I still really dislike any use of 'strong' as I think it's confusing given how prominently we used that term during Dart 2. I would much prefer 'weak checking' and 'full checking'.

Reply:
Why not keep the wording consistent with the flag?

```
To allow the null safety feature to be rolled out incrementally, we define a new mode of execution: `null-safety` mode. To enable this mode, you can pass the `--[no]null-safety` flag.

**No Null Safety** mode ...

**Null Safety** mode ...

```

Reply:
My concern with "no null safety" is that I can't discern that from Dart 2. Naming is hard! - another idea "Lax null safety" vs "enforced null safety"

Reply:
What about `Legacy` mode vs `Null Safety` mode?

Reply:
@sigmundch, why is discerning this from Dart 2 in the flag important? Isn't the point of `--no-null-safety` exactly to make it "seem like" it's still Dart 2?

@jakemac53 I don't like the term "legacy". It has a dual meaning of "from the past" and "obsolete", and the latter meaning is a problematic association for code that developers have invested much into.

Reply:
I just would like to not give the illusion that this is a feature that can be turned off. We need to effectively communicate that `--no-null-safety` will only remove the enforcement of null safety checks, but will keep around the new syntax, the actual checks, and any warnings they may produce in weak-mode.

Reply:
In light of the recent discussions of disallowing the combined mixed-mode + strong-mode, I wonder if terminology derived from the state of libraries (mixed vs not) could more clearly convey what we mean instead of using the weak/strong terms. For example:

* **mixed null-safety mode**: libraries can start using null safety, but null-safety is weakly enforced, that is, null values can flow into non-nullable typed variables.
* **pure null-safety mode**: all libraries migrated + strong null-safety enforcement

Thoughts?

Reply:
I like that. I've in other contexts been using the terms 'partial null safety' and 'full null safety'